aptools/utilities/bunch_generation.py

```diff
@@ -308,6 +316,76 @@ def _create_flattop_longitudinal_profile_with_smoothing(z_c, length, n_part,
     z = z - length/2 + z_c
     return z
 
+def _create_rectan_trapezoidal_longitudinal_profile(z_c, length, n_part, head_current,
+                                         min_len_scale_noise):
+    """ Creates a rectangular trapezoidal longitudinal profile """
+    
+    # Make sure number of particles is an integer
+    n_part = int(n_part)
+    if min_len_scale_noise is None:
+        z = rectan_trapezoid(n_part,a=z_c-length/2,b=z_c+length/2,h1=head_current)
+    else:
+        raise NotImplementedError('Noise reduction not implemented for `trapezoidal` profile.')
+    return z
```

Comment:
Here is an easy way for smoothing your profile (in this case, you do a convolution of the z trapezoidal profile with a Gaussian of size `length/10`). This would keep the number of particles and the total charge constant, but smoothen the function (affecting a bit the heights). Could you give it a try, and see if it gives what you want? You could use this for your simulations, but I am not saying this should be merged in Wake-T, Angel may prefer to keep the consistency.
```suggestion
    z += np.random.normal(0., length/10, size=n_part)
```